src/storage.rs

```diff
@@ -0,0 +1,82 @@
+use crate::block::Block;
+
+use tokio::fs::File;
+use tokio::io::{self, AsyncReadExt, AsyncWriteExt};
+
+pub const BLOCKS_DIR: &str = "./data/blocks/";
+// pub const BLOCKS_DIR: &str = "./";
+
+pub struct Storage {
+    blocks_dir_path: Option<String>,
+}
+
+impl Storage {
+    pub fn new(dir_path: Option<String>) -> Self {
+        Storage {
+            blocks_dir_path: dir_path,
+        }
+    }
+
+    pub async fn write_block_to_disk(&self, block: Block) -> io::Result<()> {
+        let mut filename = match self.blocks_dir_path.clone() {
+            Some(dir) => dir,
+            None => String::from(BLOCKS_DIR),
+        };
+
+        filename.push_str(&block.hash_as_hex());
+        filename.push_str(&".sai");
+
+        println!("{:?}", filename);
+
+        let mut buffer = File::create(filename).await?;
+
+        let byte_array: Vec<u8> = block.into();
```

Comment:
Can we please do something like block.serialize() instead of block.into()?

I'm okay with using serde/bincode for this for now, but we need to wrap it in a function somewhere, imo, even if it just does block.into() for now.